doctr/models/recognition/master/transformer_pt.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import math
+import torch
+from torch import nn
+from typing import Tuple
+
+__all__ = ['Decoder', 'positional_encoding', 'create_look_ahead_mask', 'create_padding_mask']
+
+
+def positional_encoding(position: int, d_model: int = 512) -> torch.Tensor:
+    """Implementation borrowed from this pytorch tutorial:
+    <https://pytorch.org/tutorials/beginner/transformer_tutorial.html>`_.
+
+    Args:
+        position: Number of positions to encode
+        d_model: depth of the encoding
+
+    Returns:
+        2D positional encoding as described in Transformer paper.
+    """
+    pe = torch.zeros(position, d_model)
+    position = torch.arange(0, position, dtype=torch.float).unsqueeze(1)
+    div_term = torch.exp(torch.arange(0, d_model, 2).float() * (-math.log(10000.0) / d_model))
+    pe[:, 0::2] = torch.sin(position * div_term)
+    pe[:, 1::2] = torch.cos(position * div_term)
+    return pe.unsqueeze(0)
+
+
+def create_look_ahead_mask(size: int) -> torch.Tensor:
+    # With torch transformers, True for pad and 0 False for sequences
+    mask = ~ (torch.triu(torch.ones(size, size)) == 1).transpose(0, 1)
+    return mask[:, None, :]
+
+
+def create_padding_mask(seq: torch.Tensor, padding: int = 0) -> torch.Tensor:
+    # With torch transformers, True for pad and 0 False for sequences
+    seq = ~ torch.eq(seq, padding)
+    return seq  # (batch_size, seq_len)
```

Comment:
I think it is clearer to define the function for both masks in the transformer module as they did in the tensorflow transformer tutorial, so that you just have to call both masks from there when building master model, instead of performing low-level operations specific to transformer in MASTER.

Reply:
Tell me if I'm wrong